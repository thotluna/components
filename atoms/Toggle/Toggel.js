import React from 'react'
import PropTypes from 'prop-types'
import styles from './Toggel.module.css'
import withStyles from '../../hocs/withStyles'

export const Toggel = ({ name, checked, onChange, getStyles }) => {
  return (
    <input
      className={getStyles('Toggel')}
      aria-label={name}
      type="checkbox"
      checked={checked}
      onChange={onChange}
      aria-pressed={checked}
      role="button"
    />
  )
}

Toggel.propTypes = {
  name: PropTypes.string.isRequired,
  getStyles: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,
  checked: PropTypes.bool,
}

Toggel.defaultProps = {
  checked: false,
}

export default withStyles(styles)(Toggel)
