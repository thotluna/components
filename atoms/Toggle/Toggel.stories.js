import React from 'react'
import Toggel from '.'

const Template = (args) => <Toggel {...args} />

export default {
  title: 'Atoms/Toggel',
  component: Toggel,
  args: {
    name: 'Toggel',
  },
  argTypes: {
    onChange: {
      table: {
        disable: true,
      },
    },
  },
}

export const Default = Template.bind({})

export const Active = Template.bind({})
Active.args = { isChecked: true }

export const Inactive = Template.bind({})
Inactive.args = { isChecked: false }
