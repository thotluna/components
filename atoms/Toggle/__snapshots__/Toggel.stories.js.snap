// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Atoms/Toggel Active 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <input
    aria-label="Toggel"
    aria-pressed={false}
    checked={false}
    className="Toggel"
    role="button"
    type="checkbox"
  />
</div>
`;

exports[`Storyshots Atoms/Toggel Default 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <input
    aria-label="Toggel"
    aria-pressed={false}
    checked={false}
    className="Toggel"
    role="button"
    type="checkbox"
  />
</div>
`;

exports[`Storyshots Atoms/Toggel Inactive 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <input
    aria-label="Toggel"
    aria-pressed={false}
    checked={false}
    className="Toggel"
    role="button"
    type="checkbox"
  />
</div>
`;
